packages/components/data-table-manager/src/column-settings-manager/column-settings-manager.tsx

```diff
@@ -61,7 +95,7 @@ export const handleColumnsUpdate = (
     const columns = isSwap ? selectedColumns : availableColumns;
     const draggedColumn = columns.find(
       (col) => col.key === dragResult.draggableId
-    );
+    )!;
```

Comment:
For what I understood, here we're searching in the available columns for the one dragged by `id` so I think we should always find it.
Once it is found, it call the `onUpdateColumns` callback to set the new columns order.

In case we would want to be more defensive for any reason I might not be seeing, I think we could ignore the update if we don't find the column.
What do you think?
```
const draggedColumn = columns.find(
  (col) => col.key === dragResult.draggableId
)!;

// push the column in the new position if draggedColumn is found
if (draggedColumn) {
    onUpdateColumns([
      ...items.slice(0, dragResult.destination.index),
      draggedColumn,
      ...items.slice(dragResult.destination.index),
    ]);
}
```

Reply:
If it's not too much effort, I think we can maybe be more on the safe side?

Reply:
Sure.

Updated in this commit c3696bfee231a5ffc8eaaf06d4b1ec0034c72d86